kobo-uncaged/device/types.go

```diff
@@ -40,210 +38,79 @@ type firmwareVersion struct {
 
 // KuOptions contains some options that are required
 type KuOptions struct {
-	PreferSDCard bool
-	PreferKepub  bool
-	PasswordList []string
-	Thumbnail    thumbnailOption
+	PreferSDCard         bool
+	PreferKepub          bool
+	PasswordList         []string
+	EnableDebug          bool
+	AddMetadataByTrigger bool
+	Thumbnail            thumbnailOption
 }
 
 // Kobo contains the variables and methods required to use
 // the UNCaGED library
 type Kobo struct {
-	Device          koboDevice
+	Device          kobo.Device
 	fw              firmwareVersion
 	KuConfig        *KuOptions
 	DBRootDir       string
 	BKRootDir       string
 	ContentIDprefix cidPrefix
 	useSDCard       bool
-	MetadataMap     map[string]KoboMetadata
+	MetadataMap     map[string]uc.CalibreBookMeta
 	UpdatedMetadata []string
+	BooksInDB       map[string]bool
```

Comment:
If you want to save memory here, you can use an empty struct, which has a size of 0.

```suggestion
	BooksInDB       map[string]struct{}
```

Reply:
I forgot about that! Thanks.

Reply:
Although, map[string]bool is nice to work with...

Reply:
Yes, that's true, so in general, that's what I use unless it's a large map (like this one) or in one used often (also this one).

To check it, it's just `if _, ok := whatever[key]; ok {` rather than `if whatever[key] {`. To set it, it's just `whatever[key] = struct{}{}` rather than `whatever[key]`.

Reply:
To be honest, KU isn't designed to be particularly memory efficient in some ways. The main metadata map for example. But yeah, may as well trim the fat where possible.

I shudder to think how it might handle a library with 10's of thousands of books!